plugins/filter_aws/aws.c

```diff
@@ -102,17 +100,33 @@ static void expose_aws_meta(struct flb_filter_aws *ctx)
                     "aws." FLB_FILTER_AWS_HOSTNAME_KEY,
                     ctx->hostname);
     }
+
+    for (i = 0; i < ctx->tags_count; i++) {
+        /* len("aws.ec2.tags.") = 13 */
+        env_tag_name = flb_sds_create_size(13 + ctx->tag_keys_len[i]);
+        if (!env_tag_name) {
+            flb_errno();
+            continue;
+        }
+        env_tag_name = flb_sds_printf(&env_tag_name, "aws.ec2.tags.%s", ctx->tag_keys[i]);
+        if (!env_tag_name) {
+            flb_errno();
+            continue;
+        }
+        flb_env_set(env, env_tag_name, ctx->tag_values[i]);
+        flb_sds_destroy(env_tag_name);
+    }
 }
 
 static int cb_aws_init(struct flb_filter_instance *f_ins,
                        struct flb_config *config,
                        void *data)
 {
-    int use_v2;
+    int imds_version = FLB_AWS_IMDS_VERSION_2;
     int ret;
     struct flb_filter_aws *ctx = NULL;
+    struct flb_filter_aws_init_options *options = data;
```

Comment:
Note: it assumes that `data` argument is provided as `NULL` by default. I also introduced the `options` struct on my own, so would be great to get your opinion if it is in line with the style of the codebase in general.

Reply:
What is your thinking on introducing the options struct- what problem does it solve? What is the reasoning behind this specific struct being for the two fields I see it currently contains vs just storing options on the context like the current code does for all options?